kafka/metrics.go

```diff
@@ -40,6 +41,7 @@ const (
 	msgProducedCountKey             = "producer.messages.count"
 	msgProducedBytesKey             = "producer.messages.bytes"
 	msgProducedUncompressedBytesKey = "producer.messages.uncompressed.bytes"
+	messageWriteLatencyKey          = "producer.messages.write.latency"
```

Comment:
This metric name doesn't seem right. `OnBrokerWrite` is issued per broker, and data is written to the Kafka broker at any time.

- https://pkg.go.dev/github.com/twmb/franz-go/pkg/kgo#HookBrokerWrite

For this reason, we should probably change the metric name to something like `messaging.kafka.write.latency`, since it may or may not be producing records for the request.

`OnBrokerWrite` does receive the request key metadata, so we could use the string name for the key ID and add it as a dimension to `OnBrokerWrite` metrics, see:

- https://pkg.go.dev/github.com/twmb/franz-go/pkg/kmsg#Produce

Reply:
`if key == int16(kmsg.Produce) ` condition was added in L399 so that we only record `produce` write latency. What I understood from the original issue was to filter out all the other requests sent to broker but record only produce latency. But if we are interested in broker latency in general, happy to make the change.

Reply:
Ah, that is the case; I had forgotten about the issue's contents; thanks for reminding me. It seems to have been the original intent, but considering it a bit more, it seems more future-proof and feature-complete to record latency using the stringified operation key as the dimension. This allows us to visualize more closely if/when Kafka write operations take a long time or if they fail as well, not only the produce operation.

It would also be beneficial to then open a follow-up issue to also cover https://pkg.go.dev/github.com/twmb/franz-go/pkg/kgo#HookBrokerRead.

What do you think?

Reply:
I see, that sounds good 👍 I will update the PR to address this.